box_connector.py

```diff
@@ -0,0 +1,245 @@
+            data = list(data)[0]
+            file_path = data.get('path')
+            file = open(file_path, "rb")
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "There was an issue opening the file: " + str(e.message))
+
+        result = self.box._create_file(name=name, parent_id=folder_id, file=file)
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+
+        action_result.add_data(result)
+        summary = {}
+        if result.get('type') != 'error' and result.get('entries'):
+            entries = result.get('entries')[0]
+            summary['file_id'] = str(entries.get('id'))
+            summary['sha1'] = entries.get('sha1')
+            summary = action_result.update_summary(summary)
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "There was an issue with the file creation: " + str(result))
+        # Add a dictionary that is made up of the most important values from data into the summary
+        # BaseConnector will create a textual message based off of the summary dictionary
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully uploaded file")
+
+    def _handle_create_folder(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the platform
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        name = param["folder_name"]
+        parent_id = param["parent_id"]
+
+        result = self.box._create_folder(name=name, parent_id=parent_id)
+
+        action_result.add_data(result)
+        summary = {}
+
+        if result.get('type') != 'error':
+            summary['folder_id'] = str(result.get('id'))
+            summary = action_result.update_summary(summary)
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "There was an issue with the folder creation: " + str(result))
+        # Add a dictionary that is made up of the most important values from data into the summary
+        # BaseConnector will create a textual message based off of the summary dictionary
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully created folder")
+
+    def handle_action(self, param):
+
+        ret_val = phantom.APP_SUCCESS
+
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id == 'test_connectivity':
+            ret_val = self._handle_test_connectivity(param)
+
+        elif action_id == 'upload_file':
+            ret_val = self._handle_upload_file(param)
+
+        elif action_id == 'create_folder':
```

Comment:
Would it be possible to consider adding `delete_folder` and `delete_file` here as well?
This would greatly help adding tests for these actions. We can chain the test to create -> delete. This would help avoid running into duplicated folders later on.

Reply:
This app is a community app and as we are not adding playbook and automation for the community app. So, we are not consider to add `delete_folder` and `delete_file` actions.

Reply:
That makes sense. Thank you for your explanation.